rustworkx-core/src/generators/random_graph.rs

```diff
@@ -619,14 +620,141 @@ where
+///     Threshold when ``beta`` is ``None``.
+/// * `seed` - An optional seed to use for the random number generator.
+/// * `default_node_weight` - A callable that will return the weight to use
+///     for newly created nodes.
+/// * `default_edge_weight` - A callable that will return the weight object
+///     to use for newly created edges.
+///
+/// # Example
+/// ```rust
+/// use rustworkx_core::petgraph;
+/// use rustworkx_core::generators::hyperbolic_random_graph;
+///
+/// let g: petgraph::graph::UnGraph<(), ()> = hyperbolic_random_graph(
+///     &vec![0.4, 2., 3.],
+///     &vec![0., 0., 0.],
+///     None,
+///     2.,
+///     None,
+///     || {()},
+///     || {()},
+/// ).unwrap();
+/// assert_eq!(g.node_count(), 3);
+/// assert_eq!(g.edge_count(), 2);
+/// ```
+pub fn hyperbolic_random_graph<G, T, F, H, M>(
+    radii: &[f64],
+    angles: &[f64],
+    beta: Option<f64>,
+    r: f64,
+    seed: Option<u64>,
+    mut default_node_weight: F,
+    mut default_edge_weight: H,
+) -> Result<G, InvalidInputError>
+where
+    G: Build + Create + Data<NodeWeight = T, EdgeWeight = M> + NodeIndexable + GraphProp,
+    F: FnMut() -> T,
+    H: FnMut() -> M,
+    G::NodeId: Eq + Hash,
+{
+    let num_nodes = radii.len();
+    if num_nodes == 0 || radii.len() != angles.len() {
+        return Err(InvalidInputError {});
+    }
+    if radii.iter().any(|x| x.is_infinite() || x.is_nan()) {
+        return Err(InvalidInputError {});
+    }
+    if angles.iter().any(|x| x.is_infinite() || x.is_nan()) {
+        return Err(InvalidInputError {});
+    }
+    if radii.iter().fold(0_f64, |a, &b| a.min(b)) < 0. {
+        return Err(InvalidInputError {});
+    }
+    if angles.iter().map(|x| x.abs()).fold(0_f64, |a, b| a.max(b)) > PI {
+        return Err(InvalidInputError {});
+    }
+    if beta.is_some_and(|b| b < 0. || b.is_nan()) {
+        return Err(InvalidInputError {});
+    }
+    if r < 0. || r.is_nan() {
+        return Err(InvalidInputError {});
+    }
+
+    let mut rng: Pcg64 = match seed {
+        Some(seed) => Pcg64::seed_from_u64(seed),
+        None => Pcg64::from_entropy(),
+    };
+    let mut graph = G::with_capacity(num_nodes, num_nodes);
+    if graph.is_directed() {
+        return Err(InvalidInputError {});
+    }
+
+    for _ in 0..num_nodes {
+        graph.add_node(default_node_weight());
+    }
+
+    let between = Uniform::new(0.0, 1.0);
+    for (v, (r1, theta1)) in radii
+        .iter()
+        .zip(angles.iter())
+        .enumerate()
+        .take(num_nodes - 1)
+    {
+        for (w, (r2, theta2)) in radii.iter().zip(angles.iter()).enumerate().skip(v + 1) {
+            let dist = hyperbolic_distance(r1, theta1, r2, theta2);
+            let is_edge = match beta {
+                Some(b) => {
+                    let prob = 1. / ((b / 2. * (dist - r)).exp() + 1.);
+                    let u: f64 = between.sample(&mut rng);
+                    u < prob
```

Comment:
Higly recommend us to expand this comparison instead of caculating the reciprocal of a large f64

Reply:
Good catch, this is fixed.